test/test_mps.py

```diff
@@ -8196,7 +8204,7 @@ def helper(dtype):
 
         [helper(dtype) for dtype in [torch.float32, torch.float16, torch.int32, torch.int16, torch.uint8, torch.int8]]
```

Comment:
Don't fail the entire test, just skip int32 if you are on `product_version < 14.0` here

Reply:
Done. Unfortunately the test fails on all int types for pre14, so added the skip for those to the test as well as into torch check in the mps kernel implementation.